src/db.rs

```diff
@@ -38,315 +40,410 @@ impl Default for Connection {
+pub async fn get_connection(uuid: Uuid) -> Result<Connection> {
+    let uuidstr = uuid.as_hyphenated().to_string();
 
-    let mqstr = "SELECT allowed_digests, allowed_policies, min_fw_api_major, min_fw_api_minor, allowed_build_ids FROM policy WHERE id = ? AND valid = 1";
-    let polval: Vec<(String, String, u32, u32, String)> = trnsx.exec(mqstr, (pid,))?;
-    Ok(policy::Policy {
-        allowed_digests: serde_json::from_str(&polval[0].0).unwrap(),
-        allowed_policies: serde_json::from_str(&polval[0].1).unwrap(),
-        min_fw_api_major: polval[0].2,
-        min_fw_api_minor: polval[0].3,
-        allowed_build_ids: serde_json::from_str(&polval[0].4).unwrap(),
+    let dbpool = get_dbpool().await?;
+
+    let query_str = "SELECT policy, fw_api_major, fw_api_minor, fw_build_id, launch_description, fw_digest FROM conn_bundle WHERE id = ?";
+    let new_query_str = replace_binds(dbpool.any_kind(), query_str);
+
+    let con_row = sqlx::query(&new_query_str)
+        .bind(uuidstr)
+        .fetch_one(&dbpool)
+        .await?;
+    Ok(Connection {
+        policy: con_row.try_get::<i32, _>(0)? as u32,
+        fw_api_major: con_row.try_get::<i32, _>(1)? as u32,
+        fw_api_minor: con_row.try_get::<i32, _>(2)? as u32,
+        fw_build_id: con_row.try_get::<i32, _>(3)? as u32,
+        launch_description: con_row.try_get::<String, _>(4)?,
+        fw_digest: con_row.try_get::<String, _>(5)?,
     })
 }
 
-pub fn delete_policy(pid: &u64) -> Result<()> {
-    let mut dbconn = get_dbconn()?;
+pub async fn delete_connection(uuid: Uuid) -> Result<Uuid> {
+    let uuidstr = uuid.as_hyphenated().to_string();
 
-    let mqstr = "DELETE from policy WHERE id = ?";
+    let dbpool = get_dbpool().await?;
 
-    let mut trnsx = dbconn.start_transaction(TxOpts::default())?;
-    trnsx.exec_drop(mqstr, (&pid,))?;
-    trnsx.commit()?;
-    Ok(())
+    let query_str = "DELETE from conn_bundle WHERE id = ?";
+    let new_query_str = replace_binds(dbpool.any_kind(), query_str);
+
+    sqlx::query(&new_query_str)
+        .bind(uuidstr)
+        .execute(&dbpool)
+        .await?;
+    Ok(uuid)
 }
 
-pub fn delete_connection(uuid: Uuid) -> Result<Uuid> {
-    let mut dbconn = get_dbconn()?;
+pub async fn insert_policy(policy: &policy::Policy) -> Result<u64> {
+    let allowed_digests_json = serde_json::to_string(&policy.allowed_digests)?;
+    let allowed_policies_json = serde_json::to_string(&policy.allowed_policies)?;
+    let allowed_build_ids_json = serde_json::to_string(&policy.allowed_build_ids)?;
 
-    let mqstr = "DELETE from conn_bundle WHERE id = ?";
+    let dbpool = get_dbpool().await?;
 
-    let mut trnsx = dbconn.start_transaction(TxOpts::default())?;
-    trnsx.exec_drop(mqstr, (uuid.as_hyphenated().to_string(),))?;
-    trnsx.commit()?;
-    Ok(uuid)
+    let db_type = env::var("KBS_DB_TYPE").expect("KBS_DB_TYPE not set");
+    let mut query_str = if db_type == "sqlite" {
```

Comment:
maybe we can just have the part that is different set conditionally and then format it into the larger string?